src/aims_ui/database_resources/login_as_user.py

```diff
@@ -0,0 +1,32 @@
+import os
+import sqlite3
+import hashlib, binascii, os
+
+
+def authenticate_user(input_username, input_password):
+  """Authenticate password given username"""
+  def get_pass(uname):
+    """Get passwordfor a user, given the username"""
+    for hashed_password in cur.execute(
+        'SELECT password FROM USERS WHERE username=?', (uname, )):
+      return hashed_password[0]
+
+  def verify_password(stored_password, provided_password):
+    """Verify a stored password against one provided by user"""
+    salt = stored_password[:64]
+    stored_password = stored_password[64:]
+    pwdhash = hashlib.pbkdf2_hmac('sha512', provided_password.encode('utf-8'),
+                                  salt.encode('ascii'), 100000)
+    pwdhash = binascii.hexlify(pwdhash).decode('ascii')
+    return pwdhash == stored_password
+
+  con = sqlite3.connect(os.path.join(os.path.dirname(__file__), 'auth.db'))
```

Comment:
It seems like this will re-open the database each time a user is authenticated, which is pretty expensive in the grand scheme of things.

It'd be better to open a database connection when the app starts-up, and store it inside the app instance for later use.

Reply:
Working on this but it's due to be replaced by an active directory handling implementation

Reply:
Complete, have added a connection method which is initiated at the start and called on when needed, then cleaned up on app halt